go-controller/pkg/util/ipsec.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func GetAllEstablishedIPsecTunnels() (sets.Set[string], error) {
+	ipsecTunnels := sets.Set[string]{}
+	stdout, stderr, err := RunIPsec("showstates")
+	if err != nil {
+		return ipsecTunnels, fmt.Errorf("failed to retrieve ipsec states %v: %v", stderr, err)
```

Comment:
nit;; stderr can come right after string "states " might be confusing

Reply:
changes the error message accordingly.